tests/appsec/test_automated_user_and_session_tracking.py

```diff
@@ -0,0 +1,186 @@
+        assert self.r_user_blocked.status_code == 403
+
+    def setup_user_blocking_sdk(self):
+        self.config_state_1 = rc.rc_state.reset().apply()
+        self.config_state_2 = rc.rc_state.set_config(*BLOCK_USER).apply()
+        self.r_user_blocked = weblog.get("user/me?sdk_user=sdkUser")
+
+        rc.rc_state.reset().apply()  # cleanup RC
+
+    def test_user_blocking_sdk(self):
+        assert self.config_state_1[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.config_state_2[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        interfaces.library.assert_waf_attack(self.r_user_blocked, rule="block-users")
+        assert self.r_user_blocked.status_code == 403
+
+
+@rfc("https://docs.google.com/document/d/1Z0kQ5f_GZR5thtwQjC6C_iY9C85Q-Cf-vFj9h_xd0zE")
+@features.user_monitoring
+class Test_Automated_Session_Tracking:
+    def setup_session_tracking(self):
+        self.r_create_session = weblog.get("session/new")
+        self.cookies = self.r_create_session.cookies
+        self.r_user = weblog.get("session/user?sdk_user=sdkUser", cookies=self.cookies,)
+
+    def test_session_tracking(self):
+        assert self.r_create_session.status_code == 200
+        assert self.r_user.status_code == 200
+
+        for _, trace, span in interfaces.library.get_spans(request=self.r_user):
+            meta = span.get("meta", {})
+            assert "usr.session_id" in meta
+
+
+BLOCK_SESSION = (
+    "datadog/2/ASM_DD/rules/config",
+    {
+        "version": "2.1",
+        "metadata": {"rules_version": "1.2.6"},
+        "rules": [
+            {
+                "id": "block-sessions",
+                "name": "Block Session Addresses",
+                "tags": {"type": "block_user", "category": "security_response"},
+                "conditions": [
+                    {
+                        "parameters": {"inputs": [{"address": "usr.session_id"}], "data": "blocked_sessions"},
+                        "operator": "exact_match",
+                    }
+                ],
+                "transformers": [],
+                "on_match": ["block"],
+            }
+        ],
+        "rules_data": [{"id": "blocked_sessions", "type": "data_with_expiration", "data": []},],
+    },
+)
+
+
+@rfc("https://docs.google.com/document/d/1Z0kQ5f_GZR5thtwQjC6C_iY9C85Q-Cf-vFj9h_xd0zE")
+@features.user_monitoring
+@scenarios.appsec_runtime_activation
+class Test_Automated_Session_Blocking:
+    def setup_session_blocking(self):
+        self.config_state_1 = rc.rc_state.reset().apply()
+        self.r_create_session = weblog.get("session/new")
+        self.cookies = self.r_create_session.cookies
+        BLOCK_SESSION[1]["rules_data"][0]["data"].append({"value": self.r_create_session.text, "expiration": 0})
+        self.config_state_2 = rc.rc_state.set_config(*BLOCK_SESSION).apply()
+        self.r_session_blocked = weblog.get("session/user?sdk_user=sdkUser", cookies=self.cookies,)
```

Comment:
do we really need the sdk_user arg ? this is only session blocking no ? i haven't read the whole session blocking RFC yet so maybe i'm wrong on this one
```suggestion
        self.r_session_blocked = weblog.get("session/user", cookies=self.cookies,)
```

Reply:
Nope, the problem is that the endpoint is used in the session fingerprint too, and there it's needed (or at least was when the tests was written).

Reply:
ok idk, feel free to close the comment if you think i'm wrong

Reply:
Lest's resolve this comment as the new version has removed this endpoints.